aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageSubPathStrategyListTest.kt

```diff
@@ -0,0 +1,205 @@
+            third = RandomTempFile(THIRD_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(THIRD_FILE_PATH, third, StorageUploadFileOptions.defaultInstance())
+            fourth = RandomTempFile(FOURTH_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(FOURTH_FILE_PATH, fourth, StorageUploadFileOptions.defaultInstance())
+            fifth = RandomTempFile(FIFTH_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(FIFTH_FILE_PATH, fifth, StorageUploadFileOptions.defaultInstance())
+
+            customFile = RandomTempFile(CUSTOM_FILE_NAME, SMALL_FILE_SIZE)
+            synchronousStorage.uploadFile(CUSTOM_FILE_PATH, customFile, StorageUploadFileOptions.defaultInstance())
+        }
+    }
+
+    @After
+    fun tearDown() {
+        synchronousStorage.remove("photos/2023/$FIRST_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/2023/$SECOND_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/2024/$THIRD_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/2024/$FOURTH_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/$FIFTH_FILE_NAME", StorageRemoveOptions.defaultInstance())
+        synchronousStorage.remove("photos/$CUSTOM_FILE_NAME", StorageRemoveOptions.defaultInstance())
+    }
+
+    @Test
+    fun testListWithIncludeStrategyAndStoragePath() {
+        val path = StoragePath.fromString("public/photos/")
+        val options = AWSS3StoragePagedListOptions
+            .builder()
+            .setPageSize(10)
+            .setSubpathStrategy(SubpathStrategy.Include)
+            .build()
+
+        val result = synchronousStorage.list(path, options)
+
+        result.items.mapNotNull { it.path }.apply {
+            Assert.assertEquals(6, size)
+            Assert.assertTrue(contains("public/photos/05"))
+            Assert.assertTrue(contains("public/photos/2023/01"))
+            Assert.assertTrue(contains("public/photos/2023/02"))
+            Assert.assertTrue(contains("public/photos/2024/03"))
+            Assert.assertTrue(contains("public/photos/2024/04"))
+            Assert.assertTrue(contains("public/photos/202$/custom"))
+        }
+    }
+
+    @Test
+    fun testListWithExcludeStrategyAndStoragePath() {
+        val options = AWSS3StoragePagedListOptions
+            .builder()
+            .setPageSize(10)
+            .setSubpathStrategy(SubpathStrategy.Exclude())
+            .build()
+
+        var result = synchronousStorage.list(StoragePath.fromString("public/photos/"), options)
+
+        result.items.mapNotNull { it.path }.apply {
+            Assert.assertEquals(1, size)
+            Assert.assertTrue(contains("public/photos/05"))
+        }
+
+        result.excludedSubpaths.apply {
```

Comment:
These list assertions can be improved using kotest assertions (may need to add `androidTestImplementation(libs.test.kotest.assertions)` to the build.gradle.kts if it's not there).

```kotlin
result.excludedSubpaths shouldContainExactly listOf("public/photos/2023/", "public/photos/2024/", "public/photos/202$/")
```

Reply:
that's pretty neat, unit tests update to use kotest.